src/chat.rs

```diff
@@ -3580,14 +3591,30 @@ pub(crate) async fn update_chat_contacts_table(
     id: ChatId,
     contacts: &HashSet<ContactId>,
 ) -> Result<()> {
+    let now = time();
```

Comment:
Maybe pass the "sent" timestamp explicitly to this function? At least this way the result doesn't depend on the time when the message is received and is the same on all devices